examples/pigment-css-vite-ts/src/components/ColorSchemeProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from 'react';
+
+const ColorSchemeContext = React.createContext<{
+  colorScheme: string;
+  setColorScheme: React.Dispatch<React.SetStateAction<string>>;
+}>({
+  colorScheme: 'light',
+  setColorScheme: () => '',
+});
+
+function setCookie(name: string, value: string, days: number = 100) {
+  let expires = '';
+  if (days) {
+    const date = new Date();
+    date.setTime(date.getTime() + days * 24 * 60 * 60 * 1000);
+    expires = `; expires=${date.toUTCString()}`;
+  }
+  document.cookie = `${name}=${value || ''}${expires}; path=/`;
+}
+
+export function ColorSchemeProvider({
+  colorScheme: initialColorScheme,
+  children,
+}: React.PropsWithChildren<{ colorScheme: string }>) {
+  const [colorScheme, setColorScheme] = React.useState<string>(initialColorScheme);
+
+  const contextValue = React.useMemo(
+    () => ({ colorScheme, setColorScheme }),
+    [colorScheme, setColorScheme],
+  );
+
+  // Set the colorScheme in localStorage
+  React.useEffect(() => {
+    setCookie('colorScheme', colorScheme);
```

Comment:
Regardless of whether it runs completely client side, I wanted to remember the preference the user had, this is why I kept this logic (e.g. switch to dark mode, refresh the page, it's still dark mode).

Reply:
Ah I see what you mean, refactor so that the initial value does not depend on cookie. Let me know if this is better in your opinion. I agree, we should simplify the example as much as possible 👌